File: Medium/2026-08-16-2029-Stone-Game-IX/solution.js
var stoneGameIX = function(stones) {
    let count = [0, 0, 0];
    
    for (let stone of stones) {
        count[stone % 3]++;
    }
    
    if (count[0] % 2 === 1) {
        return count[1] > 0 || count[2] > 1;
    } else {
        return count[1] > count[2] || (count[1] > 0 && count[2] > 0);
    }
};

File: Medium/2026-08-16-2029-Stone-Game-IX/solution.java
class Solution {
    public boolean stoneGameIX(int[] stones) {
        int count0 = 0, count1 = 0, count2 = 0;
        
        for (int stone : stones) {
            if (stone % 3 == 0) count0++;
            else if (stone % 3 == 1) count1++;
            else count2++;
        }
        
        if (count1 == 0 && count2 == 0) return false;
        if (count1 == 0) return count0 % 2 == 0;
        if (count2 == 0) return count0 % 2 == 0;
        
        return count1 > count2;
    }
}